### PR TITLE
Create flyte execution

### PR DESCRIPTION
**Pull request by ckiosidis**
# Hey, I just made a Pull Request!

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [x] All tests pass
- [x] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [x] This PR has NO breaking change to public API
- [x] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteExecution.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+  /**
+   * ex:<project>:<domain>:<name>.
+   * @param urn
+   * @return
+   */
+  public static FlyteExecution fromUrn(String urn) {
+    final String[] splitted = urn.split(":");
+    Preconditions.checkArgument(splitted.length == 4);
+    Preconditions.checkArgument(splitted[0].equals("ex"));
```
Comment (narape) [inline styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteExecution.java:53]: Please add unit tests that make this preconditions fail

File: styx-scheduler-service/src/test/java/com/spotify/styx/test/FlyteExecutionTest.java [hunk 2]
```diff
@@ -57,4 +59,21 @@ public void testFromUrn() {
...
+    exceptionRule.expectMessage("Expected 4 parts in URN.");
+    FlyteExecution.fromUrn("ex:flyte-test:testing");
+  }
+
+  @Test
+  public void testFromUrlWrongPrefix() {
+    exceptionRule.expect(IllegalArgumentException.class);
+    exceptionRule.expectMessage("Expected URN to start with ex.");
+    FlyteExecution.fromUrn("lp:flyte-test:testing:test-from-url-wrong-prefix");
+  }
```
Comment (narape) [inline styx-scheduler-service/src/test/java/com/spotify/styx/test/FlyteExecutionTest.java:78]: In recent versions of junit `ExpectedException.none()` have been deprecated in favor of using `Assert.assertThrows()`
```suggestion

  @Test
  public void testFromUrnMalformedString() {
    var exception = Assert.assertThrows(
        IllegalArgumentException.class,
        () -> FlyteExecution.fromUrn("ex:flyte-test:testing")
    );

    assertThat(exception.getMessage(), is("Expected 4 parts in URN."));
  }

  @Test
  public void testFromUrlWrongPrefix() {
    var exception = Assert.assertThrows(
        IllegalArgumentException.class,
        () -> FlyteExecution.fromUrn("lp:flyte-test:testing:test-from-url-wrong-prefix")
    );

    assertThat(exception.getMessage(), is("Expected URN to start with ex."));
  }
```

**Review by narape — APPROVED** [review]
I made a small suggestion. Is up to you if you want to merge it like it is now or take the suggestion.

File: styx-scheduler-service/src/test/java/com/spotify/styx/test/FlyteExecutionTest.java (hunk 2 above)
Comment (ckiosidis) [reply styx-scheduler-service/src/test/java/com/spotify/styx/test/FlyteExecutionTest.java:78]: assertThrows is introduced in junit5. 
The project is still o junit4 https://github.com/spotify/styx/blob/master/pom.xml#L474
I can look into it another time

**Pull request merged**